server/src/main/java/io/spine/server/bus/DeadMessageHandler.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.bus;
+
+import io.spine.annotation.Internal;
+import io.spine.core.MessageEnvelope;
+
+/**
+ * An interface defining a function which handles a dead message.
+ *
+ * <p>If a message with no target dispatchers found is passed to the bus, it will result in a call
+ * to this function. The function produces {@link MessageUnhandled} instance describing
+ * the dead message. It may also process the given message (e.g. store it into the bus store).
+ *
+ * @author Dmytro Dashenkov
+ */
+@Internal
+public interface DeadMessageHandler<E extends MessageEnvelope<?>> {
```

Comment:
Also, let's discuss the naming. We have handlers in `bus` world already. And that's not the one (at least, judging on the super-interface, which is absent). 

From the top of my head, that's `DeadMessageTrap`, which rhymes with the ["Wire Trap" pattern definition](http://www.enterpriseintegrationpatterns.com/patterns/messaging/WireTap.html). In which case, its method should be `capture(...)`